server/src/main/java/org/opensearch/gateway/ShardsBatchGatewayAllocator.java

```diff
@@ -153,6 +153,8 @@ public void validate(TimeValue timeValue) {
     protected final ConcurrentMap<String, ShardsBatch> batchIdToStoreShardBatch = ConcurrentCollections.newConcurrentMap();
     private final TransportNodesListGatewayStartedShardsBatch batchStartedAction;
     private final TransportNodesListShardStoreMetadataBatch batchStoreAction;
+    private Set<ShardId> timedOutPrimaryShardIds;
+    private Set<ShardId> timedOutReplicaShardIds;
```

Comment:
We clear out these Set whenever the `innerAllocateUnassignedBatch` starts its execution. Otherwise tests for this would become complicated. Let me know if this helps

Reply:
Updated